modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/url/HtmlLinkProcessor.java

```diff
@@ -278,4 +282,31 @@ private Map<String, String> extractUrlParams( final String urlQuery )
             split( query.replace( "&amp;", "&" ) );
     }
 
+    private String addQueryParamsIfPresent( final String url, final String urlQuery )
+    {
+        if ( urlQuery == null )
+        {
+            return url;
+        }
+        final String query = urlQuery.startsWith( "?" ) ? urlQuery.substring( 1 ) : urlQuery;
+
+        final StringBuilder urlSuffix = new StringBuilder();
+        final Map<String, String> queryParamsAsMap = extractUrlParams( query );
+
+        if ( queryParamsAsMap.containsKey( "query" ) )
+        {
+            urlSuffix.append( "?" ).append( decodeParameterValue( queryParamsAsMap.get( "query" ) ) );
+        }
+        if ( queryParamsAsMap.containsKey( "fragment" ) )
+        {
+            urlSuffix.append( "#" ).append( decodeParameterValue( queryParamsAsMap.get( "fragment" ) ) );
```

Comment:
you should check that query does not contain illegal characters after `decodeParameterValue( queryParamsAsMap.get( "fragment" )`